core/src/subgraph/registrar.rs

```diff
@@ -214,10 +214,25 @@ where
                     .collect::<Result<HashSet<SubgraphDeploymentId>, _>>()
             })
             .and_then(move |subgraph_ids| {
-                let provider = provider.clone();
-                stream::iter_ok(subgraph_ids).for_each(move |id| {
-                    start_subgraph(id, &*provider, logger.clone()).map_err(|()| unreachable!())
-                })
+                // This operation should finish only after all subgraphs are
+                // started. We wait for the spawned tasks to complete by giving
+                // each a `sender` and waiting for all of them to be dropped, so
+                // the receiver terminates without receiving anything.
+                let (sender, receiver) = tokio::sync::mpsc::channel::<()>(1);
```

Comment:
Why not a oneshot? (I'm assuming those are a little cheaper to create, which may not be true.)

Reply:
The oneshot sender is not clonable, so we'd have to wrap it an `Arc`. The approaches seem equivalent to me so I'll just merge this as is.